test/unit/guard.cc

```diff
@@ -28,15 +28,17 @@ stack_break_f(char *ptr)
 	return sum;
 }
 
-static char stack_buf[SIGSTKSZ];
-
 static int
 main_f(va_list ap)
 {
 	stack_t stack;
-	stack.ss_sp = stack_buf;
-	stack.ss_size = SIGSTKSZ;
 	stack.ss_flags = 0;
+	stack.ss_size = SIGSTKSZ;
+	stack.ss_sp = malloc(stack.ss_size);
```

Comment:
Nit: please use `xmalloc` and drop the check for NULL.

Reply:
This comment is still unresolved. Did you miss it or is there anything wrong with xmalloc that prevents it from being used here? Overall, we are shifting towards xmalloc throughout the code base, see #3534.

Also, please don't resolve comments by yourself. Instead reply to a comment and let the reviewer resolve it, otherwise it's difficult for the reviewer to figure out what you did. If you fully agree, it suffices to reply with something like "Fixed."

Reply:
> Did you miss it or is there anything wrong with xmalloc that prevents it from being used here?

Manual check for NULL lets me write that the allocation failed to reject file. So if allocations fails reject file contains message "cannot allocate memory for alt signal stack" while it will fail without any message if I use xmalloc. As for me, the first variant is better.

Reply:
Changed to xmalloc.